liquibase-core/src/main/java/liquibase/resource/PathHandlerFactory.java

```diff
@@ -95,32 +98,21 @@ public Resource getResource(String resourcePath, boolean includeResourceAccessor
      */
     public OutputStream openResourceOutputStream(String resourcePath, boolean includeResourceAccessor, boolean createIfNotExists) throws IOException {
         Resource resource = getResource(resourcePath, includeResourceAccessor);
-        if (resource == null) {
+        if (!resource.exists()) {
             if (createIfNotExists) {
                 return createResource(resourcePath);
             } else {
                 return null;
             }
         }
-        return resource.openOutputStream();
+        return resource.openOutputStream(createIfNotExists);
     }
 
     /**
-     *
-     * Given a path to a resource, return true if this is an absolute path or false if not
-     *
-     * @param  path       The path to consider
-     * @return boolean    True if this is an absolute path and false if not
+     * Adapts a resource found by the PathHandlerFactory to the ResourceAccessor interface so it can be used
+     * with the standard "duplicate file logic" in the ResourceAccessors
      *
      */
-    public boolean isAbsolute(String path) throws IOException {
-        PathHandler plugin = getPlugin(path);
-        if (plugin == null) {
-            throw new IOException("Cannot parse resource location: '" + path + "'");
-        }
-        return plugin.isAbsolute(path);
-    }
-
     private static class FoundResourceAccessor implements ResourceAccessor {
```

Comment:
It would be good to have a description of this inner class.

Reply:
There is a description for this class, must have been moved or something due to the way the diff is rendered, but it's definitely there.